src/components/cards/CreatePool/TokenPrices.vue

```diff
@@ -0,0 +1,112 @@
+<script lang="ts" setup>
+import usePoolCreation from '@/composables/pools/usePoolCreation';
+import useBreakpoints from '@/composables/useBreakpoints';
+import useNumbers from '@/composables/useNumbers';
+import useTokens from '@/composables/useTokens';
+
+import { computed } from 'vue';
+
+type Props = {
+  toggleUnknownPriceModal: () => void;
+};
+
+defineProps<Props>();
+
+/**
+ * COMPOSABLES
+ */
+const { upToLargeBreakpoint } = useBreakpoints();
+const { tokensList } = usePoolCreation();
+const { tokens, priceFor, injectedPrices } = useTokens();
+const { fNum } = useNumbers();
+
+/**
+ * COMPUTED
+ */
+const validTokens = computed(() => tokensList.value.filter(t => t !== ''));
+const knownTokens = computed(() =>
+  validTokens.value.filter(t => priceFor(t) !== 0 && !injectedPrices.value[t])
+);
+const unknownTokens = computed(() =>
+  validTokens.value.filter(t => priceFor(t) === 0 || injectedPrices.value[t])
+);
+const hasUnknownPrice = computed(() =>
+  validTokens.value.some(t => priceFor(t) === 0)
+);
+</script>
+
+<template>
+  <BalCard noPad shadow="none" :noBorder="upToLargeBreakpoint">
+    <div class="p-4 dark:border-gray-600 border-b" v-if="!upToLargeBreakpoint">
+      <BalStack horizontal spacing="sm" align="center">
+        <h6 class="dark:text-gray-300">
+          {{ $t('tokenPrices') }}
+        </h6>
+        <BalTooltip class="mt-1" :text="$t('correctTokenPricing')" />
+      </BalStack>
+    </div>
+    <div class="p-2 px-4">
+      <BalStack vertical isDynamic spacing="sm">
+        <BalStack
+          v-for="token in knownTokens"
+          :key="`tokenPrice-known-${token}`"
+          justify="between"
+        >
+          <span>{{ tokens[token].symbol }}</span>
+          <BalStack horizontal justify="center">
+            <div>
+              <div class="-mr-1">
+                <span>{{ fNum(priceFor(token), 'usd') }}</span>
+              </div>
+            </div>
+            <img
+              class="h-5"
+              :src="require('@/assets/images/icons/coingecko.svg')"
+            />
+          </BalStack>
+        </BalStack>
+      </BalStack>
+      <BalStack
+        vertical
+        isDynamic
+        spacing="xs"
+        :class="{ 'mt-2': knownTokens.length }"
+      >
+        <button
+          @click="toggleUnknownPriceModal"
+          :class="[
+            'hover:text-blue-500',
+            {
+              'text-red-500': hasUnknownPrice
+            }
+          ]"
+          v-for="token in unknownTokens"
+          :key="`tokenPrice-unknown-${token}`"
+        >
+          <BalStack horizontal isDynamic justify="between">
+            <span
+              :class="{ 'font-medium': injectedPrices[token] === undefined }"
```

Comment:
can this be null? because you can check for both null/undefined using double equal `== null`

Reply:
sure

Reply:
actually I think I prefer the undefined check here.

Reply:
or we could do a falsy check